extensions/vault/runtime/src/main/java/io/quarkus/vault/config/VaultServerRuntimeConfig.java

```diff
@@ -0,0 +1,172 @@
+package io.quarkus.vault.config;
+
+import static io.quarkus.vault.internal.LogConfidentialityLevel.LOW;
+
+import java.time.Duration;
+import java.util.Map;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+import io.quarkus.vault.internal.LogConfidentialityLevel;
+
+@ConfigRoot(name = "vault", phase = ConfigPhase.RUN_TIME)
+public class VaultServerRuntimeConfig {
+
+    public static final String DEFAULT_K8S_TOKEN_PATH = "/var/run/secrets/kubernetes.io/serviceaccount/token";
+    public static final String DEFAULT_KV_SECRET_ENGINE_PATH = "secret";
+    public static final String KV_SECRET_ENGINE_VERSION_V1 = "1";
+    public static final String KV_SECRET_ENGINE_VERSION_V2 = "2";
+    public static final String DEFAULT_RENEW_GRACE_PERIOD = "1H";
+    public static final String DEFAULT_CACHE_PERIOD = "10M";
+    public static final String K8S_CACERT = "/var/run/secrets/kubernetes.io/service-account/ca.crt";
+    public static final String DEFAULT_CONNECT_TIMEOUT = "5S";
+    public static final String DEFAULT_READ_TIMEOUT = "1S";
+    public static final String DEFAULT_TLS_SKIP_VERIFY = "false";
+
+    /**
+     * vault url. eg https://localhost:8200
+     */
+    @ConfigItem
+    public Optional<String> url;
```

Comment:
I think this would be better as an `Optional<URI>`.

Or even a `URI` as I suppose there's a default value? In this case the default value should be defined in the `ConfigItem` with `defaultValue`. This way, it will be properly documented.

Reply:
this is vault's url. it has to be a url, or else I am missing something. can you elaborate?
are you sure that is appropriate to put a default on the url? the only thing we can do is set http://localhost:8200 to cover the "[Running Vault for Development](https://hub.docker.com/_/vault)". but I do not like too much putting a default with no tls. and this is covering a very tiny use case.
what do you think?